packages/react/src/components/SidePanel/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import Heading from '../Heading';
+import IconButton from '../IconButton';
+import Overlay from '../Overlay';
+
+const SidePanel = ({
+  children,
+  className,
+  footer,
+  isOpen: isOpenProp,
+  onClose,
+  title,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = React.useState(isOpenProp);
+
+  const toggleIsOpen = () => setIsOpen(!isOpen);
+
+  const handleClose = () => {
+    toggleIsOpen();
+    onClose();
```

Comment:
What if `onClose === undefined`?

Reply:
Oh, just saw the defaults. Yeah that's nifty.
Although it builds an useless method in memory :P

Reply:
Yeah it's fine 🙂